### DISCUSSION TITLE
nextjs server stops a few seconds after running "npm run dev"

### DISCUSSION THREAD

**Discussion by TazApps** [vercel/next.js#47670] [Help]
### Summary

I have created a new project. I use the tutorial to create next js with react and tried to run with "npm run dev". 
The app started 
`ready - started server on 0.0.0.0:3000, url: http://localhost:3000 `
but a few seconds later stops with no errors.
node version v19.7.0
next version 13.2.4

tried also uninstall and install node.
Thanks for the help.

**Comment by icyJoseph** [comment]
Hi,

I know you tried to reinstall Node, but could you please run this command:

```console
node -p 'process.arch'
```

Directly on your terminal. It should print out your system's architecture. I am not sure why but lately many have a 32 bits version of Node.js installed on a 64 bit architecture, and that breaks.

**Reply by timiodulate** [reply]
This helped me in solving the issue.

After verifying that the version of node I have installed is 32 bits using the command you shared, I referenced the below website to clean uninstall my current version of node:
https://reactgo.com/uninstall-node-npm-from-windows/

Then, I downloaded the 64 bits version of node from here:
https://nodejs.org/en/download

After doing the above, I deleted my node_modules folder and I reinstalled the dependencies.

Thank you very much @icyJoseph

**Reply by tigerdahuacao** [reply]
Yes, it's also working for me. In my case, I just want to upgrade my node.js from 16 to 18 since some of the package is not working any more, and accidentally choose X86 node.exe release from node.js official site, and the problem happens. After I download a X64 version for windows. It solved.

**Reply by Ferosfero1** [reply]
man you saved my day, i tried every single solution until i came up on your answer.
the version of the node was the culprit...
thanks dozens ;-)

**Reply by hbharath327** [reply]
thanks

**Comment by camerondevuk** [comment]
hello what are you using

**Comment by camerondevuk** [comment]
@icyJoseph I will take actions with this now

**Comment by 0xmarkovurnek** [comment]
I am facing the same issue and when I run the node -p 'process.arch' command I get x64. When I run npm run dev it runs for a second, generates .next file, starts the server and exits suddenly without a error. I have tried deleting .next, node modules and package lock file and installing it again. I have also tried different version of node 16,17,18 and 19 (using nvm).I have also tried restarting my pc. I have used the npx create-next-app normal and with the experimental flag and it doesn't work. Normally works on Nextjs12 and create-react-app, vite etc. Tested it on macos Catalina version 10.15.3, processor Inter Core i5.

**Reply by opulencedeveloper** [reply]
Have you fixed it yet?

**Reply by 0xmarkovurnek** [reply]
It works now after I updated my OS. Still don't know what was the issue.

**Comment by opulencedeveloper** [comment]
I have a similar issue here, I also tried to run one of my old projects and it still doesn't work

**Reply by aman-singv** [reply]
update npm to the latest version

**Comment by camerondevuk** [comment]
please send me the term if it not working.

**Comment by Konadu-Akwasi-Akuoko** [comment]
I am also running into this issue. For me every time I run `npm i` it wouldn't run and it'll say due to the network error. But I removed the nextjs folder and node_modules folder. After that I installed next js with `npm install next@latest` and since then mine too is not working. It starts but after a few seconds it'll die.

**Comment by yacine-yac** [comment]
Is there any solution to this problem

**Reply by Konadu-Akwasi-Akuoko** [reply]
Well, I downgraded to nextjs `13.4.1`, and it is working now, try it and see

**Reply by dorienh** [reply]
Downgrading did not work for me. Nor did deleting .next/ or reinstalling node. Just using template next site.

**Comment by abrarfahim19** [comment]
Try deleting the node modules and reinstalling with `npm i`. It worked for me.

**Reply by teo009** [reply]
This simple solution just worked for me :,D

**Reply by furyu-chikama** [reply]
Worked well for me as well, thanks. lol

**Reply by King-OJ** [reply]
This worked for me. Apparently 'npm i' is the command to install node modules for next app instead of 'npm install'

**Reply by arpittiwari24** [reply]
Thanks . It worked for me

**Reply by call0fcode** [reply]
This also worked for me. I just deleted the "node_modules" folder manually, reinstall the deps with `npm i` and it suddenly started to work again.

**Sidenote**: In my case, not just `npm run dev` but also `npm run build` was not working (it starts building but exits after the first console log saying "Creating an optimized production build ...").

**Comment by kelvinsekx** [comment]
Updating my OS worked for me.
If you use windows or wsl, go to your settings and download updates.

It works 🍰

**Comment by ugraraz** [comment]
I think if you have any duplicate routes in your pages/api folder. It might cause this issue. for example: pages/api/test.ts and pages/api/test/index.ts. 

In my case, It happened to me.

**Comment by perisicnikola37** [comment]
If someone is still wondering about the solution, just delete the "node_modules" folder.

In Linux:
`sudo rm -rf node_modules`

**Comment by tzr0125** [comment]
I have a similar issue , I use`npm install next@latest` and it solved.
My nextjs version is 14.0.2 before I upgrade and now it is 14.1.4.
And I still dont know why

**Reply by alaato** [reply]
That worked for me for some reason
I had 14.1 and was running it on a school pc running Linux, after updating to 14.2.3, it started working.

eve tho, at home I had 14.1 and had no problems

**Comment by ahsan131-hub** [comment]
I had to delete the .next, node_modules folder, update the npm version to the latest and reinstall packages. then it worked.
